From 26527787abf4615186d90e1b5c089fbdd722aa17 Mon Sep 17 00:00:00 2001
From: Peter Delevoryas <pdel@meta.com>
Date: Tue, 13 Sep 2022 16:35:00 -0700
Subject: [PATCH 64/65] npcm8xx: Allow bios to be omitted

If the user provides the bios using `-device loader`, then the `-bios`
is *not* required.

Signed-off-by: Peter Delevoryas <peter@pjd.dev>
---
 hw/arm/npcm8xx_boards.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/hw/arm/npcm8xx_boards.c b/hw/arm/npcm8xx_boards.c
index c7d71d4df2..cfdb39efd1 100644
--- a/hw/arm/npcm8xx_boards.c
+++ b/hw/arm/npcm8xx_boards.c
@@ -39,11 +39,6 @@ static void npcm8xx_load_bootrom(MachineState *machine, NPCM8xxState *soc)
 
     filename = qemu_find_file(QEMU_FILE_TYPE_BIOS, bios_name);
     if (!filename) {
-        error_report("Could not find ROM image '%s'", bios_name);
-        if (!machine->kernel_filename) {
-            /* We can't boot without a bootrom or a kernel image. */
-            exit(1);
-        }
         return;
     }
     ret = load_image_mr(filename, machine->ram);
-- 
2.38.2

